elastic/synthetics-recorder
# feat: Asynchronously report journey test result

## Summary


part of #61 
When replaying the script, the test result is returned to the frontend after reaching the end of the journey(either success or fail). We'd like to see progressive feedback when each step is finished.




## Implementation details



When the output is returned from the Synthetics CLI, Electron parses the output and constructs event object and then sends it via `webContents.send()`. The available event types are `journey/start`, `journey/end`, and `step/end` at the moment. The frontend subscribes to the event emitter and displays relevant result information whenever available. 

## How to validate this change
click the Replay Steps button to start the test, observe the application showing feedback after each step while the test is still executed in the browser.
